pkg/api/norman/store/cluster/cluster_store.go

```diff
@@ -306,14 +315,42 @@ func loadDataFromTemplate(clusterTemplateRevision *v3.ClusterTemplateRevision, c
 			return nil, httperror.WrapAPIError(err, httperror.ServerError, "Error processing clusterTemplate answers")
 		}
 		keyParts := strings.Split(question.Variable, ".")
-		values.PutValue(dataFromTemplate, val, keyParts...)
+		if strings.HasPrefix(question.Variable, "rancherKubernetesEngineConfig.privateRegistries") {
+			// for example: question.Variable = rancherKubernetesEngineConfig.privateRegistries[0].url
+			key := keyParts[len(keyParts)-1]
```

Comment:
`key` is not needed until like 325, so I would recommend moving it just above `registry, ok := registryMap[index]`. Generally, it's a good idea to have variable declaration and usage as close as possible (there are some exceptions if we have like a whole variable declaration block, but it's not the case here).

Reply:
Fixed by moving the line to above where it is used